clang/lib/CIR/CodeGen/CIRGenStmt.cpp

```diff
@@ -459,6 +467,11 @@ mlir::LogicalResult CIRGenFunction::buildReturnStmt(const ReturnStmt &S) {
   assert(!UnimplementedFeature::requiresReturnValueCheck());
   auto loc = getLoc(S.getSourceRange());
 
+  // TODO(cir): Rewrite the logic to handle ReturnStmt inside SwitchStmt, then
+  // clean up the code below.
+  if (currLexScope->IsInsideCaseNoneStmt)
+    return mlir::success();
+
```

Comment:
> I'm opposed to return mlir::success(); because it will just silently skips something we don't know how to handle, I rather these things fails or crash, so that it's clear that they aren't implemented? What happens when you remove this return?

`buildReturnStmt()` assume there is exactly one return block in a region, and there is one region in a lexical scope, the only exceptions are switch scope, which has multiple regions. The related code is
```
    mlir::Block *getOrCreateRetBlock(CIRGenFunction &CGF, mlir::Location loc) {
      unsigned int regionIdx = 0;
      if (isSwitch())
        regionIdx = SwitchRegions.size() - 1;
      if (regionIdx >= RetBlocks.size())
        return createRetBlock(CGF, loc);
      return &*RetBlocks.back();
    }
```
So if we remove the return here, the following code will cause crash. `regionIdx`  will be -1, and we'll call `RetBlocks
.back()` with empty `RetBlocks`
```
int f(int x) {
  switch(x) {
    return 0;
  }
  return 1;
}
```
By the way, I believe the current implementation of `getOrCreateRetBlock()` about switch is incorrect and also should be solved after changing definition of `SwitchOp`.

Reply:
> By the way, I believe the current implementation of `getOrCreateRetBlock()` about switch is incorrect and also should be solved after changing definition of `SwitchOp`.

Right, we should fix the logic, not take shortcuts like returning `mlir::success()`. Can you elaborate on what do you mean by changing the definition of `SwitchOp`?

Reply:
I posted my thought in #528 to discuss it, thanks~